### ISSUE TITLE
Use Weblate for translations

### ISSUE THREAD

**Issue by comradekingu** [transmission/transmission#2538]
The feedback-loop from Transifex over to GitHub doesn't work, because there are no links to strings.
It is impossible to see multiple languages, suggestions are made based on deleted strings, the comment system is broken,
the messaging system is broken, the UI is broken, etc.
The terms and services are a further insult to injury.
It is not a good look for a libre project, and quality suffers. This is not just restricted to the translations I put care and effort into.
A move seems overdue at this point.

https://docs.weblate.org/en/latest/admin/continuous.html?highlight=webhook#github-setup
https://docs.weblate.org/en/latest/vcs.html?highlight=github#github-repositories

TL;DR Set up shop on https://hosted.weblate.org/hosting/ or self-host Weblate. Happy to help.

**Comment by ckerr** [comment]
TBH I don't know enough about the translation process to have an informed opinion on this. From some quick Googling it does look like Weblate has a nicer translator experience.

@mikedld wdyt?

**Comment by comradekingu** [comment]
Yes. When translators aren't subject to the TX terms of service, and the community isn't held hostage over negotiations over enterprise plans, everything seems to pull in the same direction.
 
https://hosted.weblate.org/projects/libretorrent/#information is doing well, and picotorrent was doing great on a self-hosted instance before seemingly lapsing on the domain recently.
There is also https://hosted.weblate.org/projects/nicotine-plus/ and maybe more I am forgetting.

All the available checks are beneficial, the ability to see more languages and context at once, and so on.
I did try to help qBittorrent while I used Transifex, and it isn't doing well on the linguistic front.
With Weblate it is easier to make source changes either directly, or with links to the actual files.

The current Transmission situation is "There is no activity the last weeks", so it can't hurt to abandon the lesser platform.

**Comment by mikedld** [comment]
One drawback of using Transifex some years ago (in 2017, when we switched to it) was the lack of support for .stringsdict format in their free offering, but they made it available 3 months later. Other than that, I can't think of anything else that was missing for me, but as well as @ckerr I'm not that involved with translation process to tell the difference.

The other thing is that we're lacking translation maintainers, and I'm not sure if Weblate will fix that; that might partially be the reason behind reduced activity, as I'm basically the only one who approves the join requests, and I'm not doing a great job at it. OTOH there's quite a lot of people with access there as it is, so my guess would be that no activity means that there's not that much to translate/fix for the languages where we do have translators.

As for integration with GitHub, that was a conscious decision on my part since the format and/or order used by Transifex when exporting via `tx pull` (and I'm guessing that would've been the format if it was to push to the repo on its own) differs from the one that's in place when importing via `tx push` after updating the files from code via `lupdate` or `xgettext` or `genstrings`, so I have a script to sync translations manually instead. I'd be happy to try out Weblate in this regard, but even then I'd be hesitant to allow a 3rd-party system write access to the repo and would rather schedule a periodic sync on our side in a more controlled environment.

See also: #100
CC:  @hanklank @piotrdrag (involved in that old discussion)

**Comment by ckerr** [comment]
TBH having a volunteer who can "maintain" it would on its own make it worth moving. Every once in awhile there are PRs or questions about translations that I don't know how to answer, e.g. https://github.com/transmission/transmission/pull/2540 says

> Some improvements in Turkish strings, that are not available through Transifex.

And I don't know enough about the domain to know _why_ these strings aren't available in Transifex.  Having someone to ping for i18n questions would be great.

**Comment by comradekingu** [comment]
@mikedld If you add "kingu" in https://hosted.weblate.org/access/transmissionbt/#users I can help.

**Comment by ckerr** [comment]
@mikedld I also would like permissions :smile_cat:

**Comment by mikedld** [comment]
Are we just planning on testing it out at this point or is it a general consensus that we need to move? Right now it looks to me as if nobody really cares as nobody else is joining the conversation, and I'd really like to get input from other translators... Maybe opening a thread on Transifex will help.

> And I don't know enough about the domain to know why these strings aren't available in Transifex.

Switching platforms won't help with that, the issue is with strings that are in the UI (*.xib) files that aren't supported by either one. What will help is finishing the switch to base internationalization (#270) which will move the strings out of *.xib files and into *.strings (and hopefully *.stringsdict if we want proper plurals) files.

**Comment by bitigchi** [comment]
Translator here. +1 for the move, but more urgently, macOS side needs the layout fixes before this I believe. I guess it would be unfriendly for non-developers having to surrender a sizeable chunk of storage to Xcode just for translating Transmission.

**Comment by comradekingu** [comment]
Not so sure "GPLv2-only" is better than "later" in https://hosted.weblate.org/projects/transmissionbt/#information even though it isn't specified one way or the other in the repo.

**Comment by mikedld** [comment]
These are not the final settings, and I'm not actively migrating anything at this time. Had to create at least one component, and specify the license, to apply for the Libre plan. But note that the approach right now is to be explicit about particular license versions we use, as seen in various source file headers, and there's no option to select multiple licenses in Weblate; for the justification on this, I believe @ckerr has the context (I remember it quite vaguely so don't want to put words into anyone's mouth).

**Comment by comradekingu** [comment]
@mikedld Work done is still licensed by its creators, not by project admins.
If it isn't specifically "only" anywhere else, I don't see a reason to use it for the translations.

>It may be used under GPLv2 (SPDX: GPL-2.0-only), GPLv3 (SPDX: GPL-3.0-only),
   or any future license endorsed by Mnemosyne LLC.

↑ Put this in the info box and all is well.

Going GPLv2-only ruins the entire concept, and locks translations to a license without a cure clause, etc.
The imported work from Transifex already has either this project license, or something specified there.

The effort itself  has already migrated. Making use of it is another matter.
I fixed all the formatting errors I could and redid the nb_NO translation.
French and Turkish has seen work by top flight translators.
Meanwhile at Transifex → "There is no activity the last weeks".

**Comment by ckerr** [comment]
> But note that the approach right now is to be explicit about particular license versions we use, as seen in various source file headers ... for the justification on this, I believe @ckerr has the context (I remember it quite vaguely so don't want to put words into anyone's mouth).

Sure. The current text of

> It may be used under GPLv2 (SPDX: GPL-2.0-only), GPLv3 (SPDX: GPL-3.0-only), or any future license endorsed by Mnemosyne LLC.

is to avoid pre-approving future licenses, but still make it easy to include them after they've been vetted. If future versions are in the spirit of earlier GPL versions I expect Mnemosyne would endorse them.

**Comment by DavidMunch** [comment]
Another translator here: Transifex's UX is not very good. I whole heartedly approve of a move, even without not knowing Weblate. It can only be better.

**Comment by pataquets** [comment]
My 2c: user of both services with preference for Weblate. Using only web interface with no other tooling. Thumbs up here, ready to help.

**Comment by atriwidada** [comment]
Translator of several projects on Transifex & Weblate. +1 for move to Weblate.

**Comment by crayxt** [comment]
Please transfer current language coordinators, if possible. New team members should be approved by coordinators before they can contribute.

**Comment by pataquets** [comment]
In my experience, approvals have caused me long delays in contributing if admins are busy or even no longer interested. I understand it might be necessary in some languages, but I'm 50/50 on it, especially languages with less contributors.

**Comment by mikedld** [comment]
From Transifex:

> Hi, Mike,
> 
> So you are moving Transmission to Weblate. Were you attracted by the fact that Weblate is open source?
> 
> Unfortunately, for translators, reviewers and language coordinators, Weblate lags in so many ways: the UI is hard to work with and slows down the translation and review process, there is no translation memory suggestions, no autofill (if I’m not mistaken), it is just a less refined l10n platform.
> 
> I’d be happy to coordinate the French (fr) translation in Weblate. Please give me the required permission to review strings in Weblate.
> 
> Will you be able to keep the current string review status from Transifex?
> 
> Kind regards,
> 
> *AO*
> French language coordinator | Localization Lab
> localizationlab.org | transifex.com/organization/otf
> 5C8F 0C0C 4812 7771 365C 262D 47DC 6187 47A9 22D2

**Comment by comradekingu** [comment]
@crayxt Language coordinators can be made reviewers on Hosted.
The reason quality suffers on TX stems from there being no functionality to ensure it, nor community.
There is a further divide between upstream, no linking to source, or transparency allowing people to even see translations without being logged in.
In turn TX has a stopgap measure that escalates missing functionality at a base level to the idea of roles.
It does prevent the inherent TX problem of drive-by translations, but only cements decisions that are bound to be as bad.
Instead of trying to coordinate a translation betwixt people working on it, now the authority is divided into finding people to entrust with the ability to keep people from doing so. I have seen that work _never_.
The changes made for nb_NO on Weblate serves as proof of how well it works for Transmission.
It is worse for small languages having one person held up asking for permission, and it is also more likely that person will go missing, making the problem worse. That does not scale well.

@mikedld 
Just to be clear, that is the user AO, not an admin from TX or similar.
Having discussed it in person, it seems to be a case of lacking experience in using Weblate.
To be clear, translations, (zen view) reviews (matrix view), coordination (explained above), leading projects and administrating and integrating (explained below), is heaps and bounds better in Weblate than TX. The inability to do things quickly, span large amounts of strings, or actually see multiple languages at once is the first showstopper.
Then the TX comment system is hidden, and the non-threaded messaging system is a pain.
If the objective is to have people not respond or reply, or find your filed issues, TX is the way to go.
In terms of integration, the tx pull/tx push is flakey, and will gladly remove all attribution from all strings.
Weblate in contrast actually credits translators down to the VCS system.
On TX it is easy to think strings are pulled and translations don't end up used, or having to recreate branches with everything attributed to the person doing it. This includes renaming locales, because that isn't supported, and TX lets people use broken or erroneous locales to make matters worse.
If it wasn't a pain to ensure quality passed for mustard before, now there is nothing to go by.
And don't actually click "review", because that locks strings.

Weblate is actually many l10n platforms, and TX is more of a datamining operation that makes uses of every single thing it can monetize. Those are not equal concerns, and they set the two apart fundamentally.
Moreover TX isn't just closed source, but an only-as-a-service offering subject to egregious terms.
In turn everyone must be 18 or above just to accept said terms.
TX also holds entire translation efforts hostage to negotiations over enterprise plan purchases.
The _only_ reason the Localization lab uses it is because they have six-figure yearly budgets to fork over because of their funding model. If the LL leadership had the technical prowess to do so and/or listened to their members (who prefer Weblate), it too would also have moved every project over. As a control group, the LL projects that use Weblate are doing vastly better.
Far more fixes sent upstream, more and better translations.
Also, it is easier to fix failing checks, because Weblate makes that semi-easy.
I did it for all the crucial strings failing in Transmission on Weblate.

In stark contrast to Weblate being libre software and actually helping similar projects, the closed TX only allows non-commercial libre software to be hosted without direct compensation. Transmission doesn't actually qualify there, as it takes donations.
TX closed their codebase, said they believe in open source, and then changed that around.

The fact of the matter is Weblate has translation memory suggestions, machine-translation suggestions, and automatic population of either, and optional sharing of those project or instance-wide.
On TX there are memory suggestions for deleted strings, and then one has to open those and go check to make sure they aren't live.
You want consistency, that is what you have to do, for every string.
Weblate actually handles fuzzy strings correctly instead of crudely removing strings. Etc, etc.
Strings can also be copied wholesale.
As for the UI, TX has less horizontal space available for the entry field, which wraps strings much sooner.
Trying to keep track of now multiple lines with more arbitrary entry points between source and translation, is no fun.
If the string is sufficiently long, now it even scrolls too. The solution is to copy one or both over to a sane editor, and then back again.
Imagine the process of reviewing many long strings with that in mind. 
Lets say you find an error and want to commit it in VCS. There is no link, and there is no direct editing.
The fix nukes all translations and users have to dig them out of a nested-menu memory, That isn't going to happen.
What happens is users either leave the longest and most complex strings untranslated, or sometimes auto-translate them.

Weblate leaves some things to be desired, but it isn't an out and out joke.

**Comment by mikedld** [comment]
https://github.com/BuddiesOfBudgie/budgie-desktop/releases/tag/v10.6

> ## Other Changes
> * Moved from Weblate to Transifex for translations

If someone has contacts there, would be interesting to know their reasons.

**Comment by comradekingu** [comment]
@mikedld Maybe @JoshStrobl knows.

**Comment by papadakospan** [comment]
I can still visit the Transifex site. HaAs it been terminated/operations moved house or not?

**Comment by ckerr** [comment]
@mikedld is this issue still something up for consideration?

I don't think it makes sense to leave this open forever; we should either consider moving or close this issue as `declined`.

**Comment by comradekingu** [comment]
@ckerr As good an idea now as it was a year ago.
10+ people agree here, among those some of the very best libre software translators around.
For the year in passing, there was a ~5-fold increase in Weblate activity on the main instance https://hosted.weblate.org/stats/
Transifex has had the same issues for 15+ years. Show me just one project that manages to avoid them.
Integration, management, team-building, communication, upstream contributions, translation, review.
For every way of interacting with it, it fails with no upside. That being said, I haven't tried the very latest developer tools, but it was a breaking and forced change, so no redeeming quality in and of itself.

Budgie moved to Transifex when it got forked out of Solus.
It seems to have been the decision of the main developer, and I have no idea as to why it was done.
https://forum.endeavouros.com/t/budgie-main-dev-leaves-solus/22104/22?page=2
There were problems importing it, the old Weblate instance wasn't advertised on https://weblate.org/discover/ and the new effort uses erroneous locales.
One year on, and the quality isn't there, with sporadic coverage. Exactly what one would expect.

https://l10n.elementary.io/languages/ (Pantheon DE) did the opposite, and quality improved greatly.
40 or so languages with full coverage. (Not counting changelog entries.)

**Comment by mikedld** [comment]
@ckerr, this dropped off my radar. I'll need some more time to consider it, let's leave open for post-4.0.0.

**Comment by maboroshin** [comment]
Transifex and Crowdin have **access control**. Transifex and Crowdin allow users to participate in translations. The advantage of Weblate is that it doesn't require users to be approved. On the other hand, this is not keep a constant quality. Users come to Weblate who use machine translation and translate "short" (video) as "shorts"(clothes). When this happens, Weblate has no access control. (Strictly speaking, the free Libre plan doesn't have this feature)

[Transifex](https://www.transifex.com/legal/terms/) and [Crowdin](https://support.crowdin.com/terms/), as well as [GitHub](https://docs.github.com/en/site-policy/github-terms/github-community-guidelines), have **terms of service that prevent harassment**. [Weblate](https://weblate.org/terms/) doesn't have this.

----

There may be one size fits all when it comes to usability. The **progress rate** from Weblate's project list is very easy to read. Other translation support sites don't have this.

Weblate's UI will erase the **search string** if there are no search hits. This is a UI that you don't see very often in many apps. [I reported this as an issue in Feb](https://github.com/WeblateOrg/weblate/issues/8833). The month of August arrived with the label "Good first issue". Sometimes you have to wait for a beginner to luck out and fix it.

Other translation assistance services **integrate a list of strings** with the details of a single string. Crowdin and Wordpress.org have automatic translation on the same page. To use automatic translation in Weblate, you must open one page per string. Weblate's list view (Zen mode) can’t display comments or automatic translations.

**Comment by comradekingu** [comment]
TX and C prevents anyone not logged in from even browsing strings.
No service actually require users to be approved.
TX and C use access control as a stopgap way to hide its underlying flaws in trying to maintain quality.
The loss is activity, and the trend is clear.

When malice or sub-part translations happen, it is mostly down to UI for how to fix it.
In no way does TX or C even compare there. It is a longer discussion to show how.
The rest is flagged errors, and that is a lot better than being tied up in waiting for votes to have any consistency.

The libre plan isn't the same as WL the product.
There is a difference between only-as-a-service software and software you can own.

The amount and content in the terms and services of each respective service heavily favours WL.

3.5 is the point that "prevents malice" https://weblate.org/terms/
Arguably people with no ill-intent could be prevented from malice somehow, but it seems
to be a feign concept.
WL also has https://github.com/WeblateOrg/weblate/blob/main/CONTACT.md
which in my view is how far preventing malice in a text-file goes.

WL shouldn't delete search strings that return no hits, but it also should be welcoming to newcomers in offering easily fixed problems to solve.

It is a somewhat good idea to not make auto-translation easier, and you are absolutely correct about comments in Zen.
Auto-translations are possible in bulk with plugins. They are marked as such, but not as easy to distinguish as they should be.

**Comment by maboroshin** [comment]
For example, One Weblate user translates dozens of languages. Is he a genius who also speaks 30 languages?
https://hosted.weblate.org/user/makarrazin14/contributions/?page=3&limit=100

In fact, he translated "Resume" as a history of education, previous jobs, This is file transfer software lol. This is a machine translation. He also translates short videos as short pants. Weblate sells the ability to control access to these users.
https://hosted.weblate.org/translate/libretube/libretube/ja/?checksum=5cbd08025a9df8db#history

This is the biggest difference from other services. In other words, Weblate is allowing this type of user. But you don't have free charge access control. I would speak equally without embellishment, Weblate appears to be banning spam with its spam reporting and detection.

**Comment by comradekingu** [comment]
I think many of those stem from changing source strings (as is the case for nb_NO)
https://hosted.weblate.org/translate/libretube/libretube/nb_NO/?checksum=d08f3ad2dc70d609
https://hosted.weblate.org/translate/libretube/libretube/nb_NO/?checksum=44ae79803f109969
https://hosted.weblate.org/translate/libretube/libretube/nb_NO/?checksum=39f69c6429aa2dda

In translating 250k strings on Hosted Weblate, I have only come across this twice.
It would be a lot easier to direct or even ban the two users in question than to fix the inherent problems that
TX and C try to hide by various other means, like hierarchy and (hierarchy) voting.

Never have I seen a project arrive at quality translations through TX or C, and this is why nobody favours it.
There is no sense of continuous effort, and instead there is a lackluster and broken tool in the way of even making a single pass of the stringbase. Fixing upstream strings is also a massive pain.

There is a case for being able to change both all and source strings, and that pans out through everyone being able to correct mistakes. Those are most often mistakes through no ill will, and the amount of people capable of correcting them are proving their worth through selection in accruing a history of good changes and trust.

Weblate can of course also have different roles and permissions, and so can the instance Hosted Weblate.

**Comment by maboroshin** [comment]
It's a fantasy, that's all. But give real examples.

It is not a string change. The accuracy of machine translation would be around 80%. This is very different from a one in 100,000 failure.

Projects that maintain high quality translations will have translation guidelines and an approval system for participation from other native speakers. In other words, it is the opposite of Weblate. Weblate, with its complete beginners and machine translations, will frequently see mistranslations. Furthermore, there is no ban.

No, Weblate is also commercialized. Weblate's access control is only paid for. Weblate also has a paid "hierarchy"!  If troll posting machine translations again, there is no free access control. For free access control, you simply need to leave Weblate.

**Comment by comradekingu** [comment]
I can tell you who the other person is if that makes you feel any better.
Someone thinking they know every language in the world is a far
cry from a world where inaccurate machine-translations are somehow avoided.
You still aren't pointing to anything unique about any platform.

The best libre software translators in the world use Weblate. I can tell you who they are too.

Bans do happen on Hosted Weblate, but they are rare.
Instead guide people into doing better work, and first hope changing their mistakes makes it back to them via e-mails they can read. Or first and foremost get notified of inconsistencies and triggered errors.
Feedback by e-mail turned on by default, as opposed to TX or C, where the default is for every mistake someone makes to keep propagating. Translatewiki does a good job of this too.

What exactly do you think prevents anyone from posting erroneous machine translations in a system of access control?
What it really does is prevent people from fixing errors and doing work when they are free to do so.

As an example it isn't convenient to sit around and wait for the language admin to approve access two days later when you found the project you wanted to contribute to for your free weekend.

As another example, two times people have gone overboard with changing things in languages they don't know, but I endlessly fix semantic and syntactic errors in other languages.
Markup formatting and proper nouns don't change between languages.

Learn the difference between Weblate and Hosted Weblate and you will discover where the great difference to the alternatives lies. One is an offering as a service, and the other is the software.
In my world "free" means something beyond gratis. However, TX and C are neither in any department.
Only offering a service at no monetary cost to non-commercial software isn't the same as a real offering for libre software.
Weblate being commercial software doesn't mean it sells its users.
I advice you to read the terms and conditions of the services you mention.

I have written translation guidelines. If your hope is that unskilled translators both read and follow them on platforms like TX or C you are in for a surprise. Weblate at least has a field for it, but don't rely on that either.
The base problem of open access or restricting it to the abilities of a language admin is the same reliance on the platform.

Hypothetically however you are correct about translation guidelines and access control fixing things, but if you actually try to put that into the real world it isn't so easy.
Tell me what project you think of as maintaining high quality translations.

**Comment by maboroshin** [comment]
> In translating 250k+ strings on Hosted Weblate, I have only come across this twice.
> The best libre software translators in the world use Weblate. 

You bring the fantasy. Vague, ambiguous. Translators can participate on any platform. Do they participate only in Weblate?  And you are basing this on what data?

> You still aren't pointing to anything unique about any platform

You just don't recognize it.

| |Transifex|Crowdin|Weblate|
|:-:|:-:|:-:|:-:|
Place|USA|Estonia|Czech Republic
Projects|40,000<br>([2018](https://www.transifex.com/about/))|[166K](https://crowdin.com/page/about-crowdin)|1,164|
Open project|[16,000](https://explore.transifex.com/)|16,869|[987](https://hosted.weblate.org/projects/?page=10&limit=100)
Users|[0.5M](https://www.transifex.com/about/)|[2.1M](https://crowdin.com/page/about-crowdin)|?
Team|[10 peples](https://www.transifex.com/about/)|[8 Peoples](https://crowdin.com/page/about-crowdin)|[3 Peoples](https://weblate.org/about/)
Eemployees| [85](https://www.zippia.com/transifex-careers-1403904/) or [29](https://www.zoominfo.com/pic/transifex-limited/347156235)| [118](https://ee.linkedin.com/company/crowdin)|[4](https://rocketreach.co/weblate-profile_b5ebc47af42e8635)
Display of translation rate|Open by project|Open by project|All Project Tables
Integrity|
Access control | Free| Free | Paid |
participation | approval | approval | anyone
Anti-Harassment Policy|[O](https://www.transifex.com/legal/terms/)|[O](https://support.crowdin.com/terms/)|[X](https://weblate.org/terms/)
Leakage of personal information | ? | ? |  [Weblate#8451](https://github.com/WeblateOrg/weblate/issues/8451) |


* [Crowdin's Advanced Search](https://crowdin.com/projects#advanced-search) could be referenced up to page 563. It's 30 projects per page. 30*562+9=16869
* [Weblate's top page](https://weblate.org/) claims 2500 libre software projects. Weblate displays 1165 projects in HostedWeblate on [the explore page](https://weblate.org/ja/discover/). And [HostedWeblate](https://hosted.weblate.org/) shows 1164. However, 987 projects are actually available for display. There are over 350 projects on servers other than HostedWeblate.
* Weblate can display all translation rates at 1 page. Excellent. However, the list of translated strings is very poor. Weblate and Crowdin can display all strings and translation history, comments, etc. in 1 page via Ajax. 
* Codes of conduct, including anti-harassment, have been adopted in many large projects.
* Weblate has already output personal information to Github and it is difficult to erase all of it.

https://github.com/WeblateOrg/weblate/issues/779 : This is a bad example of Weblate. Weblate can't exclude machine translation which joining without approval. The Weblate administrator used Github's free ban feature to stop the appeal. And it seems that the machine translators remained. Anyone can join, no access control, no harassment prevention, Most support correspondence is returned by nijel. - This is what makes Weblate unique. Weblate is easier for complete beginners and machine translators to join.

Translations of Wordpress and its plugins maintain uniformity and quality. There are guidelines and approval to participate. [Linux Mint](https://translations.launchpad.net/linuxmint) is also so.

**Comment by comradekingu** [comment]
The features implemented doesn't amount to a bunch of access control.
That access control is next to useless in garnering quality crowd-sourced translations is not a well-kept secret.
You still haven't learnt the difference between libre software and proprietary software.
What in the world is "TeraCopy"? Find something someone has heard of.
Minecraft doesn't need 37k translators, that is just a detriment.
Looking through even the "Featured projects" and "Showcases" there is nothing there.
It is abandonware, proprietary software, stuff without activity, or projects that have moved elsewhere.
GitLab remains on C, when GitHub already has abandoned ship already.

That the software is successful doesn't mean the localization will be. Case in point, the GitLab translations suck.
What you get is a representation of how good of a tool the translation platform happens to be.
You didn't come up with a single project using C or TX successfully, and I haven't seen one yet.
There is an endless stream of projects with opposite results.

Weblate has an anti-harassment policy. Moreover, harassment is illegal. You evidently don't need a CoC for anything when the most diverse community on earth does a whole lot better than the other offerings.
Words are not violence. Swearing hasn't been a problem, nor do I see why it would be.

So Xfce uses TX and has bad results (I have worked on it, so I know).
KDE, LXQt, Ubuntu Touch, Trinity, Pantheon and so on uses Weblate with good and great results.

**Comment by jjanuszkiewicz** [comment]
I've just noticed a few bad Polish translations and wanted to contribute, but Transifex UI is completely broken. Is there any progress on the Weblate migration, or an alternative way to contribute? Would it be acceptable to just create a PR with corrections?
